compose/ui/ui/src/desktopTest/kotlin/androidx/compose/ui/platform/ApplicationAccessibilityTest.kt

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.platform
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.Button
+import androidx.compose.material.DropdownMenu
+import androidx.compose.material.DropdownMenuItem
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.Text
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.assertThat
+import androidx.compose.ui.awt.ComposeWindow
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.isEqualTo
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.layout.positionInWindow
+import androidx.compose.ui.sendMouseEvent
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntRect
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.window.Popup
+import androidx.compose.ui.window.PopupPositionProvider
+import androidx.compose.ui.window.Window
+import androidx.compose.ui.window.WindowTestScope
+import androidx.compose.ui.window.density
+import androidx.compose.ui.window.runApplicationTest
+import java.awt.Point
+import java.awt.Window
+import java.awt.event.MouseEvent
+import javax.accessibility.AccessibleComponent
+import javax.accessibility.AccessibleContext
+import kotlin.test.assertNotEquals
+import org.junit.Ignore
+import org.junit.Test
+
+@OptIn(ExperimentalMaterialApi::class)
+internal class ApplicationAccessibilityTest {
+    @Test
+    fun `popup text is accessible on hover`() = runApplicationTest {
+        lateinit var window: ComposeWindow
+        val showPopup = mutableStateOf(false)
+
+        launchTestApplication {
+            Window(onCloseRequest = {}) {
+                window = this.window
+                Button(
+                    modifier = Modifier.size(100.dp),
+                    onClick = {
+                        showPopup.value = true
+                    }
+                ) {
+                    Text("Accessible button")
+                }
```

Comment:
Why this complexity? Just `showPopup = true` to show the popup.
Also, I would split this test into two:
1. Test that popup is accessible.
2. Test that overlapping accessible elements work correctly.